src/app/content/highlights/hooks/locationChange.ts

```diff
@@ -16,16 +17,28 @@ const hookBody = (services: MiddlewareAPI & AppServices) => async() => {
     return;
   }
 
-  const highlights = await highlightClient.getHighlights({
-    perPage: 100,
-    scopeId: book.id,
-    sourceIds: [page.id],
-    sourceType: GetHighlightsSourceTypeEnum.OpenstaxPage,
-  });
+  const [highlights, totalCounts] = await Promise.all([
+    await highlightClient.getHighlights({
+      perPage: 100,
+      scopeId: book.id,
+      sourceIds: [page.id],
+      sourceType: GetHighlightsSourceTypeEnum.OpenstaxPage,
+    }),
+    await highlightClient.getHighlightsSummary({
+      scopeId: book.id,
+      sourceType: GetHighlightsSummarySourceTypeEnum.OpenstaxPage,
+    }),
+  ]);
 
   if (highlights.data) {
     dispatch(receiveHighlights(highlights.data));
   }
+
+  if (totalCounts.countsPerSource) {
+    dispatch(receiveHighlightsTotalCounts(totalCounts.countsPerSource));
+    const mergedTotalCounts = mergeHighlightsTotalCounts(book, totalCounts.countsPerSource);
+    dispatch(setHighlightsTotalCountsPerLocation(mergedTotalCounts));
```

Comment:
do you think it makes sense to store both of these? or should we just make a selector for the processed version? (selectors are memoized, so it'll still only be calculated once)

its good to maintain access to both the per page and the per location filter versions of this data though, because for pagination we'll need the first